platform/backend/repository/memory/src/main/kotlin/io/hamal/repository/memory/record/Func.kt

```diff
@@ -81,15 +84,42 @@ class MemoryFuncRepository : MemoryRecordRepository<FuncId, FuncRecord, Func>(
                         namespaceId = cmd.namespaceId,
                         name = cmd.name,
                         inputs = cmd.inputs,
-                        code = cmd.code
+                        codeId = cmd.codeId,
+                        codeVersion = cmd.codeVersion
                     )
                 )
                 (currentVersion(funcId)).also(CurrentFuncProjection::apply)
             }
         }
     }
 
-    override fun update(funcId: FuncId, cmd: FuncCmdRepository.UpdateCmd): Func {
+    override fun deploy(funcId: FuncId, cmd: DeployCmd): Func {
+        return lock.withLock {
+            if (commandAlreadyApplied(cmd.id, funcId)) {
+                versionOf(funcId, cmd.id)
+            } else {
+                val current = versionOf(funcId, cmd.id)
+
+                if (cmd.versionToDeploy !in CodeVersion(1)..current.code.version) {
+                    throw NoSuchElementException("${cmd.versionToDeploy} does not exist")
+                }
+
+                store(
+                    FuncDeploymentRecord(
+                        cmdId = cmd.id,
+                        entityId = funcId,
+                        deployment = cmd.versionToDeploy
+                    )
+                )
+
+                val temp = currentVersion(funcId)
+                CurrentFuncProjection.deploy(temp)
+                return temp
```

Comment:
should be the same as create or update
```suggestion
               return (currentVersion(funcId)).also(CurrentFuncProjection::apply)
```